plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -4751,7 +4760,7 @@ private VirtualMachineGuestOsIdentifier translateGuestOsIdentifier(String cpuArc
                             }
                         }
                     }
-                    vmResponseMap.put(name, new VmStatsEntry(Integer.parseInt(maxCpuUsage), networkReadKBs, networkWriteKBs, Integer.parseInt(numberCPUs), "vm"));
+                    vmResponseMap.put(name, new VmStatsEntry(Double.parseDouble(memkb)*1000,Double.parseDouble(guestMemusage)*1000,Double.parseDouble(memlimit)*1000, Double.parseDouble(maxCpuUsage), networkReadKBs, networkWriteKBs, Integer.parseInt(numberCPUs), "vm"));
```

Comment:
Is there a possibility that one of the values is not returned and hence can be null? calling parseDouble on a null value will throw NPE